### PR TITLE
HIVE-16490. Hive should not use getKeyProvider from DFSClient directly.

### PR DESCRIPTION
**Pull request by umamaheswararao**
https://issues.apache.org/jira/browse/HIVE-16490

### CODE REVIEW

File: shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java [hunk 1]
```diff
@@ -1223,11 +1223,31 @@ public static boolean isHdfsEncryptionSupported() {
...
-      DistributedFileSystem dfs = (DistributedFileSystem)FileSystem.get(uri, conf);
-
       this.conf = conf;
-      this.keyProvider = dfs.getClient().getKeyProvider();
       this.hdfsAdmin = new HdfsAdmin(uri, conf);
+      this.keyProvider = getKeyProvider();
+    }
+
+    private KeyProvider getKeyProvider() throws IOException {
+      if (isMethodExist(HdfsAdmin.class, "getKeyProvider")) {
```
Comment (ashutoshc) [inline shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java:1232]: this is not needed. Hive's minimum version required for Hadoop is 3.0

Comment (umamaheswararao) [reply shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java:1232]: That is great. I will remove that then. Thanks for pointing.

**Comment by umamaheswararao** [comment]
Thanks @ashutoshc for the review!. I have updated it. Please take a look. Thank you

File: shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java [hunk 2]
```diff
@@ -1223,11 +1223,10 @@ public static boolean isHdfsEncryptionSupported() {
     private final Configuration conf;
 
     public HdfsEncryptionShim(URI uri, Configuration conf) throws IOException {
-      DistributedFileSystem dfs = (DistributedFileSystem)FileSystem.get(uri, conf);
-
       this.conf = conf;
-      this.keyProvider = dfs.getClient().getKeyProvider();
       this.hdfsAdmin = new HdfsAdmin(uri, conf);
+      // HDFS-11687 added getKeyProvider method in hdfsAdmin from Hadoop-2.9.0
```
Comment (belugabehr) [inline shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java:1228]: Please remove this comment.  Keep this information in JIRA.  Thanks.

Comment (umamaheswararao) [reply shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java]: Done. Thanks

**Comment by umamaheswararao** [comment]
My latest commit has only removal of a comment: https://github.com/apache/hive/pull/1279/commits/2db5d300789bbc8325cae33eeb47840130f525d6
The patch already has green tests in previous runs.

**Pull request closed without merging**